### PR TITLE
Issue 16 file upload visual feedback and file size

### LINKED ISSUE #16: File Uploader Improvements - Visual Feedback, File Validation and Response Handler
## Purpose
To add several validation and feedback functionalities to the File Uploader.
- Progress Bar: Show upload progress for each file.
- File Validation Improvements `fileSize`, `fileType` and `fileCount`
- A restriction on uploaded file to prevent misuse.
- Success/Failure feedback redesigned with an AJAX Handler for better interaction.

## Benefit
This would enhance user feedback and prevent the uploading of large files that may cause problems with the comparison and report process. Provide an indicator of progress when uploading the files so the User knows the current state of their upload. Make the uploader more robust when it gets subjected to regular use.

### PR DESCRIPTION
**Pull request by jbwod**
This Pull Request will merge the features from Issue 16 to the main branch

## Additions
- Visual Feedback for uploading files
- Progress Indicators
- Additional Server and Client Side checks for intended uploads

## Changes
- Adjusted Style Changes
- Refactor of upload handler to use AJAX

# Related issue

- Resolve #16

### CODE REVIEW

File: app/static/styles/home.css [hunk 1]
```diff
@@ -1,25 +1,65 @@
...
+    display: block;
+    margin-bottom: 1rem;
+    transition: transform 0.3s ease-in-out, color 0.3s ease-in-out;
+    transform-origin: center;
+}
+
+.drop-zone:hover::before {
+    transform: scale(1.1);
+    color: #0056b3;
+}
```
Comment (loklokyx) [inline app/static/styles/home.css]: I'd like to help improve the code by aligning it with FontAwesome standards. Could I have your permission to update the code?

Comment (jbwod) [reply app/static/styles/home.css]: Go for it. Make any changes you need.
I get the impression styling is going to go through many iterations

Comment (loklokyx) [reply app/static/styles/home.css]: Done commit, thanks.

**Review by loklokyx — APPROVED** [review]
Looks good to me now, thanks!

File: app/forms.py [hunk 2]
```diff
@@ -1,23 +1,34 @@
...
         exts = current_app.config['UPLOAD_EXTENSIONS']
-        self.validators=[
+        max_size = current_app.config['MAX_CONTENT_LENGTH']
+        self.validators = [
             FileAllowed(
                 [ext.replace('.','') for ext in exts], 
                 message = f'Invalid File Type. Must be {", ".join(exts)}' 
-            )]
+            ),
+            FileSize(max_size=max_size, message=f'File exceeds the maximum size of {max_size // (1024 * 1024)}MB.')
```
Comment (Elijah017) [inline app/forms.py:17]: I believe this should be `max_size // (1000 * 1000)`. I think what your currently calculating is the `MiB` rather than the `MB`.

Comment (jbwod) [reply app/forms.py:17]: MB and MiB are synonymous.

But on most systems, even though sizes are calculated in MiB results still report as MB

**Review by Elijah017 — APPROVED** [review]

**Pull request merged**